internal/dao/log_options.go

```diff
@@ -114,6 +117,20 @@ func (o *LogOptions) ToLogItem(bytes []byte) *LogItem {
 	if len(bytes) == 0 {
 		return item
 	}
+
+	if item.GetTimestamp() != "" {
```

Comment:
This call is under heavy fire so we have to be careful here with allocations and extra calls. Best to parse out the log line here, extract the timestamp if present and build  the log item directly imho. As it is we index thru the log line twice.